src/pages/[collectionSlug]/[slug].astro

```diff
@@ -0,0 +1,183 @@
+---
+import {
+  fetchCustomCollectionPageBySlug,
+  createCustomCollectionPageMapper,
+  fetchSlug,
+} from "@/utilities/fetch";
+import { tryParseDateParts } from "@/utilities/dates";
+import PagesSection from "@/components/PagesSection.astro";
+import Layout from "@/layouts/Layout.astro";
+import RichText from "@/components/RichText.astro";
+import Tags from "@/components/Tags.astro";
+import Media from "@/components/Media.astro";
+import InPageNavigation from "@/components/InPageNavigation.astro";
+import PreviewReload from "@/components/PreviewReload.astro";
+import { fetchCollection } from "@/utilities/fetch";
+import { safeJsonParse } from "@/utilities/fetch/payload-fetch";
+
+const { collectionSlug, slug: pageSlug } = Astro.params;
+
+// Check if this slug matches a custom collection config
+const collectionConfig = await fetchSlug(
+  "custom-collections",
+  collectionSlug || "",
+);
+
+if (!collectionConfig) {
+  return Astro.redirect("/404");
+}
+
+// Fetch the specific page
+let data: any;
+
+if (!Astro.isPrerendered) {
+  data = await fetchCustomCollectionPageBySlug(
+    collectionConfig.id,
+    pageSlug || "",
+  );
+  if (!data) return Astro.redirect("/404");
+} else {
+  // @ts-expect-error
+  ({ data } = Astro.props);
+}
+
+// Map the page
+const mapper = createCustomCollectionPageMapper(collectionSlug || "");
+const item = mapper(data);
+const contentDate = data.contentDate
+  ? tryParseDateParts(data.contentDate)
+  : null;
+
+export async function getStaticPaths() {
+  // Fetch all custom collection configs and their pages to generate static paths
+  try {
+    const configsResponse = await fetchCollection("custom-collections");
+    const configsData = await safeJsonParse(configsResponse);
```

Comment:
Delete and rename the `configsResponses` above to `configsData` since it already gets called in the `fetchCollection`